testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/ActivemqServerAddonsTestCase.java

```diff
@@ -0,0 +1,178 @@
+import org.jboss.arquillian.container.test.api.ContainerController;
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.container.test.api.TargetsContainer;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.test.integration.common.HttpRequest;
+import org.jboss.as.test.integration.common.jms.JMSOperations;
+import org.jboss.as.test.integration.common.jms.JMSOperationsProvider;
+import org.jboss.as.test.manualmode.messaging.deployment.MessagingServlet;
+import org.jboss.as.test.manualmode.messaging.deployment.QueueMDB;
+import org.jboss.as.test.manualmode.messaging.deployment.TopicMDB;
+import org.jboss.as.test.module.util.TestModule;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ServerControl;
+
+/**
+ *
+ * @author Emmanuel Hugonnet (c) 2022 Red Hat, Inc.
+ */
+@RunAsClient()
+@RunWith(Arquillian.class)
+@ServerControl(manual = true)
+public class ActivemqServerAddonsTestCase {
+
+    private static final String DEFAULT_FULL_JBOSSAS = "default-full-jbossas";
+
+    @ArquillianResource
+    protected static ContainerController container;
+    @ArquillianResource
+    private Deployer deployer;
+
+    private final TestModule module = new TestModule("org.apache.activemq.artemis.addons", "org.apache.activemq.artemis");
+
+    private static final String LB_CLASS_ATT = "connection-load-balancing-policy-class-name";
+    private static final String QUEUE_LOOKUP = "java:/jms/DependentMessagingDeploymentTestCase/myQueue";
+    private static final String TOPIC_LOOKUP = "java:/jms/DependentMessagingDeploymentTestCase/myTopic";
+
+    private static final String QUEUE_NAME = "myQueue";
+    private static final String TOPIC_NAME = "myTopic";
+    private static final String DEPLOYMENT = "addons-deployment";
+
+    @Before
+    public void setup() throws Exception {
+        try ( ManagementClient managementClient = createManagementClient()) {
+            if (container.isStarted(DEFAULT_FULL_JBOSSAS)) {
+                container.stop(DEFAULT_FULL_JBOSSAS);
+            }
+            final JavaArchive jar = module.addResource("artemis-lb-policy.jar");
+            jar.addClasses(OrderedLoadBalancingPolicy.class);
+            System.setProperty("module.path", Paths.get("target" ,"wildfly", "modules").toAbsolutePath().toString());
```

Comment:
This is ok but the typical way we deal with this is via module.path in system-property-variables in the pom.

That's not necessarily better, it's just the usual handling that people might look for.